pstress/pstress-run.sh

```diff
@@ -1574,8 +1578,30 @@ pquery_test(){
         sleep 2
```

Comment:
does pquery_script checks if the core file is created by signal 4 or it generated because of some error in the database.  or does we only rely on mysqld error logs?

Reply:
pstress driver script detects the core file and reports it in the logs. In case of some error in database, it relies on the error logs.

Reply:
so better to put a message or don't show that it has generated core file in case of signal 4. because the user can get confused. 
Please submit a log report with signal 4 and signal 6. want to compare the test ou

Reply:
Crash logs due to Signal 6
[11:47:24] [2] ====== TRIAL #3 ======
[11:47:24] [2] Ensuring there are no relevant servers running...
[11:47:25] [2] Clearing rundir...
[11:47:25] [2] Generating new trial workdir /tmp/689237/3...
[11:47:25] [2] Copying datadir from Trial /home/mohit.joshi/test_p/689237/2 into /home/mohit.joshi/test_p/689237/3...
[11:47:25] [2] Starting mysqld. Error log is stored at /tmp/689237/3/log/master.err
[11:47:25] [2] Waiting for mysqld (pid: 7112) to fully start (note this is slow for Valgrind runs, and can easily take 35-90 seconds even on an high end server)...
[11:47:30] [2] Server started ok. Client: /home/mohit.joshi/mysql-8.0/bld_14july2020/install/bin/mysql -uroot -S/tmp/689237/3/socket.sock
[11:47:30] [2] Loading metadata from /home/mohit.joshi/test_p/689237/step_2.dll ...
[11:47:30] [2] /home/mohit.joshi/pstress_fork/pstress/../src/pstress-ps --database=test --threads=20 --queries-per-thread=100000 --logdir=/tmp/689237/3 --user=root --socket=/tmp/689237/3/socket.sock --seed 689237 --step 3 --metadata-path /home/mohit.joshi/test_p/689237/ --seconds 30 --tables 10 --records 100 --log-all-queries --log-failed-queries
[11:47:30] [2] pquery running (Max duration: 30s)...
[11:47:51] [2] Cleaning up & saving results if needed...
./pstress-run.sh: line 748:  7112 Aborted                 (core dumped) $CMD > ${RUNDIR}/${TRIAL}/log/master.err 2>&1
[11:47:55] [2] Killing the server with Signal 9
[11:47:56] [2] pquery run details: 129/4385 queries failed, (97.06% were successful)
[11:47:56] [2] mysqld coredump detected at /tmp/689237/3/data/core
[11:47:56] [2] Bug found (as per error log): Assertion failure: ha_innodb.cc:14993:0
New Error has been found in error log(s). Potentially a Bug, please investigate
[MY-011947] [InnoDB] Cannot open '/tmp/689237/3/data/ib_buffer_pool' for reading: No such file or directory

Signal 4:

[11:56:48] [0] ====== TRIAL #1 ======
[11:56:48] [0] Ensuring there are no relevant servers running...
[11:56:48] [0] Clearing rundir...
[11:56:48] [0] Generating new trial workdir /tmp/973229/1...
[11:56:48] [0] Copying datadir from template...
[11:56:48] [0] Starting mysqld. Error log is stored at /tmp/973229/1/log/master.err
[11:56:48] [0] Waiting for mysqld (pid: 8950) to fully start (note this is slow for Valgrind runs, and can easily take 35-90 seconds even on an high end server)...
[11:57:11] [0] Server started ok. Client: /home/mohit.joshi/mysql-8.0/bld_14july2020/install/bin/mysql -uroot -S/tmp/973229/1/socket.sock
[11:57:11] [0] Creating metadata randomly using random seed 973229 ...
[11:57:11] [0] /home/mohit.joshi/pstress_fork/pstress/../src/pstress-ps --database=test --threads=20 --queries-per-thread=100000 --logdir=/tmp/973229/1 --user=root --socket=/tmp/973229/1/socket.sock --seed 973229 --step 1 --metadata-path /home/mohit.joshi/test_p/973229/ --seconds 30 --tables 10 --records 100 --log-all-queries --log-failed-queries
[11:57:11] [0] pquery running (Max duration: 30s)...
[11:57:41] [0] 30s timeout reached. Terminating this trial...
[11:57:41] [0] TIMEOUT_INCREMENT option was disabled and set to 0
[11:57:41] [0] Cleaning up & saving results if needed...
[11:57:45] [0] Killing the server with Signal 4
[11:57:46] [0] pquery run details: 38/2006 queries failed, (98.11% were successful)
[11:57:46] [0] Bug found (as per error log): New Error has been found in error log(s). Potentially a Bug, please investigate
[MY-011947] [InnoDB] Cannot open '/tmp/973229/1/data/ib_buffer_pool' for reading: No such file or directory
[11:57:46] [0] Copying rundir from /tmp/973229/1 to /home/mohit.joshi/test_p/973229/1

Reply:
why it is showing bug found with signal 4?

Reply:
mysqld was built using the suicide patch due to which there were errors in the error log. Even if signal 4 is used, if there are genuine errors found, it will be reported

Reply:
build without the crash patch. and upload patch here.

Reply:
[14:25:08] [0] ====== TRIAL #1 ======
[14:25:08] [0] Ensuring there are no relevant servers running...
[14:25:08] [0] Clearing rundir...
[14:25:08] [0] Generating new trial workdir /tmp/296109/1...
[14:25:08] [0] Copying datadir from template...
[14:25:08] [0] Starting mysqld. Error log is stored at /tmp/296109/1/log/master.err
[14:25:08] [0] Waiting for mysqld (pid: 20263) to fully start (note this is slow for Valgrind runs, and can easily take 35-90 seconds even on an high end server)...
[14:25:14] [0] Server started ok. Client: /home/mohit.joshi/mysql-8.0/bld_06july2020/install/bin/mysql -uroot -S/tmp/296109/1/socket.sock
[14:25:14] [0] Creating metadata randomly using random seed 296109 ...
[14:25:14] [0] /home/mohit.joshi/pstress_fork/pstress/../src/pstress-ps --database=test --threads=20 --queries-per-thread=100000 --logdir=/tmp/296109/1 --user=root --socket=/tmp/296109/1/socket.sock --seed 296109 --step 1 --metadata-path /home/mohit.joshi/test_p/296109/ --seconds 60 --tables 10 --records 100 --log-all-queries --log-failed-queries
[14:25:14] [0] pquery running (Max duration: 60s)...
[14:26:14] [0] 60s timeout reached. Terminating this trial...
[14:26:14] [0] TIMEOUT_INCREMENT option was disabled and set to 0
[14:26:14] [0] Cleaning up & saving results if needed...
[14:26:18] [0] Killing the server with Signal 4
[14:26:19] [0] pquery run details: 111/3703 queries failed, (97.00% were successful)
[14:26:19] [0] Bug found (as per error log): 
[14:26:19] [0] Copying rundir from /tmp/296109/1 to /home/mohit.joshi/test_p/296109/1
[14:26:19] [1] ====== TRIAL #2 ======
[14:26:19] [1] Ensuring there are no relevant servers running...
[14:26:19] [1] Clearing rundir...
[14:26:19] [1] Generating new trial workdir /tmp/296109/2...
[14:26:19] [1] Copying datadir from Trial /home/mohit.joshi/test_p/296109/1 into /home/mohit.joshi/test_p/296109/2...
[14:26:20] [1] Starting mysqld. Error log is stored at /tmp/296109/2/log/master.err
[14:26:20] [1] Waiting for mysqld (pid: 20945) to fully start (note this is slow for Valgrind runs, and can easily take 35-90 seconds even on an high end server)...

Reply:
still it is showing Bug found (as per error log):

Reply:
Yes, "Bug found" must not be seen unless there is an actual error seen in the logs. There was a problem in the code which I have rectified. After the new changes, the coredump generated by Sig4 will be mentioned explicitly in the logs so that the user do not confuse it will other cores. Please find the logs for both Sig4 and Sig9 below

Sig 4:
[06:37:58] [0] ====== TRIAL #1 ======
[06:37:58] [0] Ensuring there are no relevant servers running...
[06:37:58] [0] Clearing rundir...
[06:37:58] [0] Generating new trial workdir /tmp/007289/1...
[06:37:58] [0] Copying datadir from template...
[06:37:58] [0] Starting mysqld. Error log is stored at /tmp/007289/1/log/master.err
[06:37:58] [0] Waiting for mysqld (pid: 615) to fully start (note this is slow for Valgrind runs, and can easily take 35-90 seconds even on an high end server)...
[06:38:03] [0] Server started ok. Client: /home/mohit.joshi/mysql-8.0/bld_06july2020/install/bin/mysql -uroot -S/tmp/007289/1/socket.sock
[06:38:03] [0] Creating metadata randomly using random seed 007289 ...
[06:38:03] [0] /home/mohit.joshi/pstress_fork/pstress/../src/pstress-ps --database=test --threads=20 --queries-per-thread=100000 --logdir=/tmp/007289/1 --user=root --socket=/tmp/007289/1/socket.sock --seed 007289 --step 1 --metadata-path /home/mohit.joshi/test_p/007289/ --seconds 50 --tables 10 --records 100 --log-all-queries --log-failed-queries
[06:38:03] [0] pquery running (Max duration: 50s)...
[06:38:53] [0] 50s timeout reached. Terminating this trial...
[06:38:53] [0] TIMEOUT_INCREMENT option was disabled and set to 0
[06:38:53] [0] Cleaning up & saving results if needed...
[06:38:57] [0] Killing the server with Signal 4
[06:38:58] [0] pquery run details: 145/4095 queries failed, (96.46% were successful)
[06:38:58] [0] mysqld coredump detected due to SIGNAL(kill -4) at /tmp/007289/1/data/core
[06:38:58] [0] Copying rundir from /tmp/007289/1 to /home/mohit.joshi/test_p/007289/1
[06:38:58] [1] ====== TRIAL #2 ======
[06:38:58] [1] Ensuring there are no relevant servers running...
[06:38:58] [1] Clearing rundir...
[06:38:58] [1] Generating new trial workdir /tmp/007289/2...
[06:38:58] [1] Copying datadir from Trial /home/mohit.joshi/test_p/007289/1 into /home/mohit.joshi/test_p/007289/2...
[06:38:59] [1] Starting mysqld. Error log is stored at /tmp/007289/2/log/master.err
[06:38:59] [1] Waiting for mysqld (pid: 1222) to fully start (note this is slow for Valgrind runs, and can easily take 35-90 seconds even on an high end server)...
./pstress-run.sh: line 958:   615 Illegal instruction     (core dumped) $CMD > ${RUNDIR}/${TRIAL}/log/master.err 2>&1
[06:39:30] [1] Server started ok. Client: /home/mohit.joshi/mysql-8.0/bld_06july2020/install/bin/mysql -uroot -S/tmp/007289/2/socket.sock
[06:39:30] [1] Loading metadata from /home/mohit.joshi/test_p/007289/step_1.dll ...
[06:39:30] [1] /home/mohit.joshi/pstress_fork/pstress/../src/pstress-ps --database=test --threads=20 --queries-per-thread=100000 --logdir=/tmp/007289/2 --user=root --socket=/tmp/007289/2/socket.sock --seed 007289 --step 2 --metadata-path /home/mohit.joshi/test_p/007289/ --seconds 50 --tables 10 --records 100 --log-all-queries --log-failed-queries
[06:39:30] [1] pquery running (Max duration: 50s)...
[06:40:21] [1] 50s timeout reached. Terminating this trial...
[06:40:21] [1] TIMEOUT_INCREMENT option was disabled and set to 0
[06:40:21] [1] Cleaning up & saving results if needed...
[06:40:25] [1] Killing the server with Signal 4
[06:40:26] [1] pquery run details: 103/4605 queries failed, (97.76% were successful)
[06:40:26] [1] mysqld coredump detected due to SIGNAL(kill -4) at /tmp/007289/2/data/core
[06:40:26] [1] Copying rundir from /tmp/007289/2 to /home/mohit.joshi/test_p/007289/2

Sig 9
=====
[06:56:04] [0] ====== TRIAL #1 ======
[06:56:04] [0] Ensuring there are no relevant servers running...
[06:56:04] [0] Clearing rundir...
[06:56:04] [0] Generating new trial workdir /tmp/192115/1...
[06:56:04] [0] Copying datadir from template...
[06:56:04] [0] Starting mysqld. Error log is stored at /tmp/192115/1/log/master.err
[06:56:04] [0] Waiting for mysqld (pid: 6731) to fully start (note this is slow for Valgrind runs, and can easily take 35-90 seconds even on an high end server)...
[06:56:09] [0] Server started ok. Client: /home/mohit.joshi/mysql-8.0/bld_06july2020/install/bin/mysql -uroot -S/tmp/192115/1/socket.sock
[06:56:09] [0] Creating metadata randomly using random seed 192115 ...
[06:56:09] [0] /home/mohit.joshi/pstress_fork/pstress/../src/pstress-ps --database=test --threads=20 --queries-per-thread=100000 --logdir=/tmp/192115/1 --user=root --socket=/tmp/192115/1/socket.sock --seed 192115 --step 1 --metadata-path /home/mohit.joshi/test_p/192115/ --seconds 50 --tables 10 --records 100 --log-all-queries --log-failed-queries
[06:56:09] [0] pquery running (Max duration: 50s)...
[06:57:00] [0] 50s timeout reached. Terminating this trial...
[06:57:00] [0] TIMEOUT_INCREMENT option was disabled and set to 0
[06:57:00] [0] Cleaning up & saving results if needed...
[06:57:04] [0] Killing the server with Signal 9
./pstress-run.sh: line 748:  6731 Killed                  $CMD > ${RUNDIR}/${TRIAL}/log/master.err 2>&1
[06:57:05] [0] pquery run details: 72/3113 queries failed, (97.69% were successful)
[06:57:05] [0] Saving full trial outcome (as SAVE_TRIALS_WITH_CORE_OR_VALGRIND_ONLY=0 and so trials are saved irrespective of whether an issue was detected or not)
[06:57:05] [0] Copying rundir from /tmp/192115/1 to /home/mohit.joshi/test_p/192115/1
[06:57:05] [1] ====== TRIAL #2 ======
[06:57:05] [1] Ensuring there are no relevant servers running...
[06:57:05] [1] Clearing rundir...
[06:57:05] [1] Generating new trial workdir /tmp/192115/2...
[06:57:05] [1] Copying datadir from Trial /home/mohit.joshi/test_p/192115/1 into /home/mohit.joshi/test_p/192115/2...
[06:57:06] [1] Starting mysqld. Error log is stored at /tmp/192115/2/log/master.err
[06:57:06] [1] Waiting for mysqld (pid: 7350) to fully start (note this is slow for Valgrind runs, and can easily take 35-90 seconds even on an high end server)...
[06:57:21] [1] Server started ok. Client: /home/mohit.joshi/mysql-8.0/bld_06july2020/install/bin/mysql -uroot -S/tmp/192115/2/socket.sock
[06:57:21] [1] Loading metadata from /home/mohit.joshi/test_p/192115/step_1.dll ...
[06:57:21] [1] /home/mohit.joshi/pstress_fork/pstress/../src/pstress-ps --database=test --threads=20 --queries-per-thread=100000 --logdir=/tmp/192115/2 --user=root --socket=/tmp/192115/2/socket.sock --seed 192115 --step 2 --metadata-path /home/mohit.joshi/test_p/192115/ --seconds 50 --tables 10 --records 100 --log-all-queries --log-failed-queries
[06:57:21] [1] pquery running (Max duration: 50s)...
[06:58:11] [1] 50s timeout reached. Terminating this trial...
[06:58:11] [1] TIMEOUT_INCREMENT option was disabled and set to 0
[06:58:11] [1] Cleaning up & saving results if needed...
[06:58:15] [1] Killing the server with Signal 9
./pstress-run.sh: line 748:  7350 Killed                  $CMD > ${RUNDIR}/${TRIAL}/log/master.err 2>&1
[06:58:16] [1] pquery run details: 141/4099 queries failed, (96.56% were successful)
[06:58:16] [1] Saving full trial outcome (as SAVE_TRIALS_WITH_CORE_OR_VALGRIND_ONLY=0 and so trials are saved irrespective of whether an issue was detected or not)
[06:58:16] [1] Copying rundir from /tmp/192115/2 to /home/mohit.joshi/test_p/192115/2

Reply:
could you paste what would haven if we give signal 4. but it gets actually crashed due to bug.

Reply:
Sig 4 when actual problem occurs
==========================

[07:40:31] [0] ====== TRIAL #1 ======
[07:40:31] [0] Ensuring there are no relevant servers running...
[07:40:31] [0] Clearing rundir...
[07:40:31] [0] Generating new trial workdir /tmp/004252/1...
[07:40:31] [0] Copying datadir from template...
[07:40:31] [0] Starting mysqld. Error log is stored at /tmp/004252/1/log/master.err
[07:40:31] [0] Waiting for mysqld (pid: 12797) to fully start (note this is slow for Valgrind runs, and can easily take 35-90 seconds even on an high end server)...
[07:40:52] [0] Server started ok. Client: /home/mohit.joshi/mysql-8.0/bld_14july2020/install/bin/mysql -uroot -S/tmp/004252/1/socket.sock
[07:40:52] [0] Creating metadata randomly using random seed 004252 ...
[07:40:52] [0] /home/mohit.joshi/pstress_fork/pstress/../src/pstress-ps --database=test --threads=20 --queries-per-thread=100000 --logdir=/tmp/004252/1 --user=root --socket=/tmp/004252/1/socket.sock --seed 004252 --step 1 --metadata-path /home/mohit.joshi/test_p/004252/ --seconds 60 --tables 10 --records 100 --log-all-queries --log-failed-queries
[07:40:52] [0] pquery running (Max duration: 60s)...
[07:41:31] [0] Cleaning up & saving results if needed...
./pstress-run.sh: line 748: 12797 Aborted                 (core dumped) $CMD > ${RUNDIR}/${TRIAL}/log/master.err 2>&1
[07:41:35] [0] Killing the server with Signal 4
[07:41:36] [0] pquery run details: 54/2139 queries failed, (97.48% were successful)
[07:41:36] [0] mysqld coredump detected due to SIGNAL(kill -4) at /tmp/004252/1/data/core
[07:41:36] [0] mysqld error detected in the log via search_string.sh scan
[07:41:36] [0] Bug found (as per error log): Assertion failure: ha_innodb.cc:14993:0
New Error has been found in error log(s). Potentially a Bug, please investigate
[MY-011947] [InnoDB] Cannot open '/tmp/004252/1/data/ib_buffer_pool' for reading: No such file or directory
[07:41:36] [0] Copying rundir from /tmp/004252/1 to /home/mohit.joshi/test_p/004252/1

Reply:
see if possible, remove this error in case of genuine bug.
mysqld coredump detected due to SIGNAL(kill -4) at /tmp/004252/1/data/core

Reply:
It is possible. I have added a check to see if the error logs contains the string "mysqld got signal 4". 
Sig 4
====
[10:08:17] [0] ====== TRIAL #1 ======
[10:08:17] [0] Ensuring there are no relevant servers running...
[10:08:17] [0] Clearing rundir...
[10:08:17] [0] Generating new trial workdir /tmp/842898/1...
[10:08:17] [0] Copying datadir from template...
[10:08:17] [0] Starting mysqld. Error log is stored at /tmp/842898/1/log/master.err
[10:08:17] [0] Waiting for mysqld (pid: 32704) to fully start (note this is slow for Valgrind runs, and can easily take 35-90 seconds even on an high end server)...
[10:08:41] [0] Server started ok. Client: /home/mohit.joshi/mysql-8.0/bld_14july2020/install/bin/mysql -uroot -S/tmp/842898/1/socket.sock
[10:08:41] [0] Creating metadata randomly using random seed 842898 ...
[10:08:41] [0] /home/mohit.joshi/pstress_fork/pstress/../src/pstress-ps --database=test --threads=20 --queries-per-thread=100000 --logdir=/tmp/842898/1 --user=root --socket=/tmp/842898/1/socket.sock --seed 842898 --step 1 --metadata-path /home/mohit.joshi/test_p/842898/ --seconds 60 --tables 10 --records 100 --log-all-queries --log-failed-queries
[10:08:41] [0] pquery running (Max duration: 60s)...
[10:09:41] [0] 60s timeout reached. Terminating this trial...
[10:09:41] [0] TIMEOUT_INCREMENT option was disabled and set to 0
[10:09:41] [0] Cleaning up & saving results if needed...
[10:09:45] [0] Killing the server with Signal 4
[10:09:46] [0] pquery run details: 191/5768 queries failed, (96.69% were successful)
[10:09:46] [0] mysqld coredump detected due to SIGNAL(kill -4) at /tmp/842898/1/data/core
[10:09:46] [0] mysqld error detected in the log via search_string.sh scan
[10:09:46] [0] Bug found (as per error log): New Error has been found in error log(s). Potentially a Bug, please investigate
[MY-011947] [InnoDB] Cannot open '/tmp/842898/1/data/ib_buffer_pool' for reading: No such file or directory
[10:09:46] [0] Copying rundir from /tmp/842898/1 to /home/mohit.joshi/test_p/842898/1
[10:09:46] [1] ====== TRIAL #2 ======
[10:09:46] [1] Ensuring there are no relevant servers running...
[10:09:46] [1] Clearing rundir...
[10:09:46] [1] Generating new trial workdir /tmp/842898/2...
[10:09:46] [1] Copying datadir from Trial /home/mohit.joshi/test_p/842898/1 into /home/mohit.joshi/test_p/842898/2...
[10:09:47] [1] Starting mysqld. Error log is stored at /tmp/842898/2/log/master.err
[10:09:47] [1] Waiting for mysqld (pid: 1053) to fully start (note this is slow for Valgrind runs, and can easily take 35-90 seconds even on an high end server)...
./pstress-run.sh: line 958: 32704 Illegal instruction     (core dumped) $CMD > ${RUNDIR}/${TRIAL}/log/master.err 2>&1
[10:10:16] [1] Server started ok. Client: /home/mohit.joshi/mysql-8.0/bld_14july2020/install/bin/mysql -uroot -S/tmp/842898/2/socket.sock
[10:10:16] [1] Loading metadata from /home/mohit.joshi/test_p/842898/step_1.dll ...
[10:10:16] [1] /home/mohit.joshi/pstress_fork/pstress/../src/pstress-ps --database=test --threads=20 --queries-per-thread=100000 --logdir=/tmp/842898/2 --user=root --socket=/tmp/842898/2/socket.sock --seed 842898 --step 2 --metadata-path /home/mohit.joshi/test_p/842898/ --seconds 60 --tables 10 --records 100 --log-all-queries --log-failed-queries
[10:10:16] [1] pquery running (Max duration: 60s)...
./pstress-run.sh: line 1450:  1309 Aborted                 (core dumped) $CMD > ${RUNDIR}/${TRIAL}/pquery.log 2>&1
[10:10:20] [1] Cleaning up & saving results if needed...
./pstress-run.sh: line 748:  1053 Aborted                 (core dumped) $CMD > ${RUNDIR}/${TRIAL}/log/master.err 2>&1
[10:10:24] [1] Killing the server with Signal 4
[10:10:25] [1] pquery run details:
[10:10:25] [1] mysqld coredump detected at /tmp/842898/2/data/core
[10:10:25] [1] mysqld error detected in the log via search_string.sh scan
[10:10:25] [1] Bug found (as per error log): Assertion failure: ha_innodb.cc:14993:0
New Error has been found in error log(s). Potentially a Bug, please investigate
[MY-011947] [InnoDB] Cannot open '/tmp/842898/2/data/ib_buffer_pool' for reading: No such file or directory
[10:10:25] [1] Copying rundir from /tmp/842898/2 to /home/mohit.joshi/test_p/842898/2